cmd/promtool/main.go

```diff
@@ -555,16 +590,19 @@ func checkRules(filename string) (int, []error) {
 		numRules += len(rg.Rules)
 	}
 
-	dRules := checkDuplicates(rgs.Groups)
-	if len(dRules) != 0 {
-		fmt.Printf("%d duplicate rule(s) found.\n", len(dRules))
-		for _, n := range dRules {
-			fmt.Printf("Metric: %s\nLabel(s):\n", n.metric)
-			for _, l := range n.label {
-				fmt.Printf("\t%s: %s\n", l.Name, l.Value)
+	if lintSettings.duplicateRules() {
+		dRules := checkDuplicates(rgs.Groups)
+		if len(dRules) != 0 {
+			errMessage := fmt.Sprintf("%d duplicate rule(s) found.\n", len(dRules))
+			for _, n := range dRules {
+				errMessage += fmt.Sprintf("Metric: %s\nLabel(s):\n", n.metric)
+				for _, l := range n.label {
+					errMessage += fmt.Sprintf("\t%s: %s\n", l.Name, l.Value)
+				}
 			}
+			errMessage += "Might cause inconsistency while recording expressions"
+			return successExitCode, []error{errors.New(errMessage)}
```

Comment:
> cause the execution to return a non-zero exit code in case
those options are not met.

So should this be a non-zero exit code, or is this a lint check that still returns a success?

Reply:
I used the same pattern as in the error condition above, but there is also a `lintErrExitCode` that we can return.
https://github.com/prometheus/prometheus/blob/53ac9d6d666acaebbdb8a004b51958f3147b1bd5/cmd/promtool/main.go#L548-L551

However, it seems that this value is ignored when errors are returned from `checkRules` because it indicates how many rules are discovered, it's not the actual exit code:
https://github.com/prometheus/prometheus/blob/53ac9d6d666acaebbdb8a004b51958f3147b1bd5/cmd/promtool/main.go#L303-L319

So maybe we should just use `0` in the case the linter fails?

Reply:
Looking further, `lintErrExitCode` is only used in `CheckMetrics`. Do we want to use it in the rules linter as part of this PR?

Reply:
Thanks! I like the approach.